how-tos/create-graphql-mutation.md

```diff
@@ -0,0 +1,232 @@
+ * @return {Promise<Object>} CreateSomethingPayload
+ */
+export default async function createSomething(parentResult, { input }, context) {
+  const { clientMutationId = null } = input;
+  // TODO: decode incoming IDs here
+  const renameMe = await context.mutations.createSomething(context);
+  return {
+    renameMe,
+    clientMutationId
+  };
+}
+```
+
+Make adjustments to the resolver function so that it reads and passes along the parameters correctly. The general pattern is:
+- Destructure `input`. Include `clientMutationId`.
+- Decode any opaque IDs that are in the input object
+- Call `context.mutations.<mutationName>` (your new plugin mutation) with the necessary arguments, and `await` a response.
+- Return an object that contains the `clientMutationId` and the object returned by the plugin mutation. (This must match the "Payload" type from the schema.)
+
+## Step 8: Register the resolver
+
+In `resolvers/Mutation/index.js` in the plugin, import your mutation resolver and add it to the default export object. Example:
+
+```js
+import createSomething from "./createSomething"
+
+export default {
+  createSomething
+};
+```
+
+If this is the first mutation for the plugin, you'll also need to import the `Mutation` object into the `resolvers` object. In `resolvers/index.js` in the plugin, import `Mutation` and add it to the default export object.
+
+```js
+import Mutation from "./Mutation"
+
+export default {
+  Mutation
+};
+```
+
+Then pass the full `resolvers` object to `registerPlugin` in the plugin's `index.js` file:
+
+```js
+import resolvers from "./resolvers";
+import schemas from "./schemas";
+
+export default async function register(app) {
+  await app.registerPlugin({
+    graphQL: {
+      resolvers,
+      schemas
+    },
+    // other props
+  });
+}
+```
+
+Calling your mutation with GraphQL should now work.
+
+## Step 9: Add a test file for your mutation resolver
+
+If your mutation resolver is in a file named `createSomething.js`, your Jest tests should be in a file named `createSomething.test.js` in the same folder. Initially you can copy and paste the following test:
+
+```js
+import createSomething from "./createSomething";
+
+test("correctly passes through to mutations.createSomething", async () => {
+  const fakeResult = { /* TODO */ };
+
+  const mockMutation = jest.fn().mockName("mutations.createSomething");
+  mockMutation.mockReturnValueOnce(Promise.resolve(fakeResult));
+  const context = {
+    mutations: {
+      createSomething: mockMutation
+    }
+  };
+
+  const result = await createSomething(null, {
+    input: {
+      /* TODO */
+      clientMutationId: "clientMutationId"
+    }
+  }, context);
+
+  expect(result).toEqual({
+    renameMe: fakeResult,
+    clientMutationId: "clientMutationId"
+  });
+});
+```
+
+This of course should be updated with tests that are appropriate for whatever your mutation resolver does. For example, verify that all ID and schema transformations happen.
```

Comment:
Why? Why does it hurt to specify that you should write tests?

Reply:
Mostly because I feel that we don't need to tell people that. Giving a example implies that they can modify it according to their needs. Anyway not a big deal.